src/state/ModalContext.tsx

```diff
@@ -2,19 +2,23 @@ import { createContext, useState } from "react";
 import { z } from "zod";
 
 const ModalContextSchema = z.object({
```

Comment:
Could you help me understand why you're using zod here and above? Is there some need for validation that the type system doesn't cover that I'm missing?

Reply:
I'm certain that TS type system can cover every single type that zod is creating in all my files. I primarily use zod because it is a fast way to create more complex types by method chaining in zod schemas and then inferring those schemas into TS types. At this point, I want to create an e2e implementation of the project first and then worry about the specificity of how things were written when it is an appropriate time.

Reply:
Sure you can move fast and not be blocked on this, but you understand that it doesn't make much sense in the first place, right? You're using a validation library to define types instead of directly defining types, and you're never using the validation/parsing. It's more verbose and has runtime costs without really any clear advantage